templates/ecmonsen-emorypipeline-ci.yaml

```diff
@@ -0,0 +1,107 @@
+Description: Setup service account and roles for Emory pipeline lambda
+AWSTemplateFormatVersion: 2010-09-09
+
+Parameters:
+  Department:
+    Description: 'The department for this resource'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+    Default: 'SysBio'
+  Project:
+    Description: 'The name of the project that this resource is used for'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+    Default: 'AMP-AD'
+  OwnerEmail:
+    Description: 'Email address of the owner of this resource'
+    Type: String
+    AllowedPattern: '^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$'
+    ConstraintDescription: 'Must be an acceptable email address syntax(i.e. joe.smith@sagebase.org)'
+    Default: 'eva.monsen@sagebase.org'
+
+Resources:
+  # !! IMPORTANT !! - AWS API will refuse to remove users that have attached resources.
+  # Therefore you must do the following before deleting them from this file:
+  # 1. Detach or remove the following user resources: login profile, attached
+  #    MFA device, access-keys, ssh-keys, and policies.
+  # 2. Detach the user from all groups.
+  LambdaArtifactsBucket:
+    Type: 'AWS::S3::Bucket'
+    Properties:
+      Tags:
+        - Key: "Department"
+          Value: !Ref Department
+        - Key: "Project"
+          Value: !Ref Project
+        - Key: "OwnerEmail"
+          Value: !Ref OwnerEmail
+
+  CfnDeployerServiceUser:
```

Comment:
lets go with more generic names here and the rest of the file. something like ServiceUser because cloudformation will create a resource by appending your stack name to this resource name. i.e. it will create something like service-accounts-emorypipeline-ServiceUser-1F2B2A33. cloudormation appends a random number to the end of the name.

Reply:
Happy to change the names, but if I may ask, why would you like the names to be different than your example?

Reply:
actually keeping as-is will make CFN generate resource names like so.. `ecmonsen-emorypipeline-ci-CfnDeployerServiceUser-1F2B2A33` which isn't bad either. keep as is fine.  FYI, cloudformation's resource name length limit is 255 chars.